src/vm/perfinfo.cpp

```diff
@@ -0,0 +1,117 @@
+#include "common.h"
+
+#if defined(FEATURE_PERFMAP) && !defined(DACCESS_COMPILE)
+#include "perfinfo.h"
+#include "pal.h"
+
+PerfInfo::PerfInfo(int pid)
+{
+    LIMITED_METHOD_CONTRACT;
+
+    WCHAR tempPath[MAX_LONGPATH+1];
+    if (!GetTempPathW(MAX_LONGPATH, tempPath))
+    {
+        return;
+    }
+
+    SString path;
+    path.Printf("%Sperfinfo-%d.map", tempPath, pid);
+    OpenFile(path);
+}
+
+void PerfInfo::LogImage(PEFile* pFile, WCHAR* guid)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+        PRECONDITION(pFile != NULL);
+        PRECONDITION(guid != NULL);
+    } CONTRACTL_END;
+
+    SString value;
+    const SString& path = pFile->GetPath();
+    value.Printf("%S%c%S", path.GetUnicode(), sDelimiter, guid);
+
+    SString command;
+    command.Printf("%s", "ImageLoad");
+    WriteLine(command, value);
+
+}
+
+void PerfInfo::WriteLine(SString& type, SString& value)
+{
+
+    CONTRACTL    
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_PREEMPTIVE;
+        PRECONDITION(type != NULL);
+        PRECONDITION(value != NULL);
+    } CONTRACTL_END;
+
+    if (m_Stream == NULL)
+    {
+        return;
+    }
+
+    SString line;
+    line.Printf("%S%c%S%c\n",
```

Comment:
Also keep in mind that the ending semicolin is within the value part of the command, any user can choose to format the value anyway they want since every command is _one_ line. I'm not sure how this affects anybody else?